cmd/imagestream.go

```diff
@@ -47,8 +56,36 @@ func (o *ImageStreamCleanupOptions) cleanupImageStreamTags(cmd *cobra.Command, a
 		o.ImageStream = args[0]
 	}
 
+	if o.Orphan == false && o.Regex != "^[a-z0-9]{40}$" {
+		log.WithFields(log.Fields{"Orphan": o.Orphan, "Regex": o.Regex}).
+			Fatal("Missing Orphan flag")
+	}
+
+	regexp, err := regexp.Compile(o.Regex)
+	if err != nil {
+		log.WithField("regex", o.Regex).Fatal("Invalid regex.")
+	}
+
 	if o.Tag && !git.IsValidSortValue(o.Sorted) {
-		log.WithField("sort_criteria", o.Sorted).Fatal("Invalid sort criteria")
+		log.WithField("sort_criteria", o.Sorted).Fatal("Invalid sort criteria.")
+	}
+
+	if o.CommitLimit !=0 && o.Orphan == true {
+		log.WithFields(log.Fields{"CommitLimit": o.CommitLimit, "Orphan": o.Orphan}).
+			Fatal("Mutually exclusive flags")
+	}
+
+	var duration time.Time
+	if len(o.Duration) > 0 {
+		var err error
+		duration, err = tparse.ParseNow(time.RFC3339, "now-" + o.Duration)
+		if err != nil {
+			log.WithError(err).
+				WithField("duration", o.Duration).
+				Fatal("Could not parse duration.")
```

Comment:
Again, "Duration" seems wrong. It's more of a timespan or delta. Also, log the error with the flag name (`WithField("older-than", o.WhateverTheNewNameIs)` or `WithField("flag_name", "--older-than"`)